java/java/JavaParser.g4

```diff
@@ -32,9 +33,31 @@ parser grammar JavaParser;
 options { tokenVocab=JavaLexer; }
 
 compilationUnit
-    : packageDeclaration? importDeclaration* typeDeclaration* EOF
+    : (packageDeclaration? importDeclaration* typeDeclaration*)
```

Comment:
Why do you use parentheses here? They are redundant since `|` operator has the lowest priority.

```suggestion
    : packageDeclaration? importDeclaration* typeDeclaration*
```